src/main/java/com/xebialabs/overthere/winrm/WinRmClient.java

```diff
@@ -486,11 +490,19 @@ private void configureTrust(final HttpClientBuilder httpclientBuilder) throws No
 
         final TrustStrategy trustStrategy = httpsCertTrustStrategy.getStrategy();
         SSLContext sslContext = SSLContextBuilder.create().loadTrustMaterial(trustStrategy).build();
-        final HostnameVerifier hostnameVerifier = httpsHostnameVerifyStrategy.getVerifier();
+        final HostnameVerifier hostnameVerifier = getHostNameVerifier();
         final SSLConnectionSocketFactory socketFactory = new SSLConnectionSocketFactory(sslContext, hostnameVerifier);
         httpclientBuilder.setSSLSocketFactory(socketFactory);
     }
 
+    private HostnameVerifier getHostNameVerifier() {
+        HostnameVerifier hostnameVerifier = httpsHostnameVerifyStrategy.getVerifier();
+        if (isUseJumpstation() && hostnameVerifier instanceof DefaultHostnameVerifier) {
```

Comment:
do we need the instanceOf check?

Reply:
Even it goes to the NOOP verifier it will not make any difference. Let's keep the conditions as minimum as required.

Reply:
Removed it, if we used Noop verifier anyway it will always return true.